src/components/ProductCard.jsx

```diff
@@ -1,15 +1,56 @@
 import React from 'react';
+import { Link } from 'react-router-dom';
 
 export default class ProductCard extends React.Component {
-  render() {
-    const { product } = this.props;
-    const { title, price, thumbnail } = product;
+  constructor(props) {
+    super(props);
+
+    this.state = { details: false }
+
+    this.openDetails = this.openDetails.bind(this);
+    this.plainProduct = this.plainProduct.bind(this);
+    this.detailedProduct = this.detailedProduct.bind(this);
+  }
+
+  detailedProduct() {
+    const { title, price, thumbnail, attributes, id } = this.props.product;
+    return (
+      <div>
+        <div>{title}</div>
+        <div>{price}</div>
+        <img src={thumbnail} alt={title} />
+        <Link to={`/productDetails/${id}`} data-testid="product-detail-link">DETALHES</Link>
+        {
+          attributes.map(({ name, value_name, id }) => {
+            return (<p key={`${id}`}>{`${name}: ${value_name}`}</p>);
```

Comment:
oi, @pedrobth! é necessário essa interpolação com `` na propriedade `key`? a seguir, com `name` e `value_name` eu consigo ver a aplicação, mas confesso que no `key` eu não entendi. caso seja, não tem problema deixar, mas caso não seja, eu sugiro o seguinte:

```suggestion
            return (<p key={id}>{`${name}: ${value_name}`}</p>);
```